apps/golf-gps/golf-gps.js

```diff
@@ -0,0 +1,306 @@
+      buttons: {
+        "SCAN": 1,
+        "QUIT": 2
+      }
+    }).then(choice => {
+      if (choice === 1) searchCourse();
+      else load();
+    });
+  } else if (distanceCalc(lastFix.lat, lastFix.lon, lat[1], lon[1]) < 1000) {
+    Bangle.buzz();
+    E.showPrompt(courseName + "\nIs this correct?", {
+      buttons: {
+        " YES ": 1,
+        " NO ": 2
+      }
+    }).then(choice => {
+      if (choice == 1) {
+        E.showPrompt("Front or Back", {
+          title: courseName,
+          buttons: {
+            "FRONT": 1,
+            "BACK": 2
+          }
+        }).then(choice => {
+          currentHole = (choice === 1) ? 1 : 10;
+          playON = true;
+          showPlayData();
+        });
+      } else searchCourse();
+    });
+  } else searchCourse();
+}
+
+function showPlayData() {
+  let date = new Date();
+  let distanceToHole = distanceCalc(lastFix.lat, lastFix.lon, lat[currentHole], lon[currentHole]);
+  let distanceFromLast = distanceCalc(lastFix.lat, lastFix.lon, latLast, lonLast) || 0;
+
+  g.reset().clearRect(Bangle.appRect);
+
+  // distance to hole in yards
+  g.setColor(g.theme.fg).setFontAlign(1, -1).setFontDroidSansMono52();
+  g.drawString(distanceToHole > 1000 ? "000" : distanceToHole.toFixed(0), W - 2, 1);
+
+  // distance from last shot in yards
+  g.setFontDroidSansMono35().setColor('#00f').drawString(distanceFromLast.toFixed(0), W - 2, 70);
+
+  // hole number and par color(red:3, green:4, blue:5)
+  g.setColor(par[currentHole] == 3 ? '#f00' : (par[currentHole] == 4 ? '#0f0' : '#00f'));
+  g.fillRect(3, 3, 47, 47);
+  g.setColor(par[currentHole] == 4 ? '#000' : '#fff').setFontAlign(0, -1).drawString(currentHole, 24, 5);
+
+  // total shots and current shots
+  g.setColor(g.theme.fg).setFontAlign(-1, -1);
+  g.drawString(totalShots, 3, 70); // total shots
+  g.drawString(score[currentHole], 3, 128); // current shots
+  g.setFont("6x8:2").drawString('TOTAL', 3, 54);
+  g.drawString('THIS', 3, 112);
+
+  // clock
+  g.setFontAlign(1, 1).setFont("7x11Numeric7Seg:4").drawString((date.getHours() > 12 ? date.getHours() % 12 : date.getHours()) + ':' + zeroPad(date.getMinutes(), 2), W - 2, H - 4);
+  g.drawString((date.getHours() > 12 ? date.getHours() % 12 : date.getHours()) + ':' + zeroPad(date.getMinutes(), 2), W - 1, H - 3);
+
+  // battery level bar
+  g.setColor('#000').drawRect(59, H * 2 / 3 - 2, W - 5, H * 2 / 3 + 4);
+  g.drawRect(58, H * 2 / 3 - 1, W - 4, H * 2 / 3 + 5);
+  g.setColor(E.getBattery() > 15 ? '#03f' : '#f00').fillRect(60, H * 2 / 3, 60 + E.getBattery() / 100 * (W - 60), H * 2 / 3 + 3);
+
+  // hdop level indicator
+  if (lastFix.hdop < 5) g.setColor('#0f0').fillRect(60, H / 3, 96, H / 3 + 5);
+  if (lastFix.hdop < 2) g.fillRect(100, H / 3, 136, H / 3 + 5);
+  if (lastFix.hdop < 1) g.fillRect(140, H / 3, W, H / 3 + 5);
+}
+
+function finishGame() {
+  let date = new Date();
+  let saveFileContents = ' ';
+  let parTotal = 0;
+
+  Bangle.setGPSPower(0);
```

Comment:
This probably isn't turning off gps, since we don't specify the name with which we started GPS:

```suggestion
  Bangle.setGPSPower(0, "golf-gps");
```